web/src/admin/files/FileUploadForm.ts

```diff
@@ -0,0 +1,134 @@
+import { DEFAULT_CONFIG } from "#common/api/config";
+import { MessageLevel } from "#common/messages";
+
+import { Form } from "#elements/forms/Form";
+import { showMessage } from "#elements/messages/MessageContainer";
+
+import { AdminApi, AdminFileListUsageEnum } from "@goauthentik/api";
+
+import { msg } from "@lit/localize";
+import { html } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+
+@customElement("ak-file-upload-form")
+export class FileUploadForm extends Form<Record<string, unknown>> {
+    @property({ type: String })
+    usage: AdminFileListUsageEnum = AdminFileListUsageEnum.Media;
+
+    @state()
+    selectedFile?: File;
+
+    private clearFileInput() {
+        this.selectedFile = undefined;
+        const fileInput = this.shadowRoot?.querySelector<HTMLInputElement>("#file-input");
+        if (fileInput) {
+            fileInput.value = "";
+        }
+    }
+
+    async send(): Promise<void> {
+        if (!this.selectedFile) {
+            throw new Error(msg("Please select a file"));
+        }
+
+        // Validate filename contains only safe characters (a-zA-Z0-9._-)
+        const safeFilenameRegex = /^[a-zA-Z0-9._-]+$/;
+        if (!safeFilenameRegex.test(this.selectedFile.name)) {
+            throw new Error(
+                msg("Filename can only contain letters, numbers, dots, hyphens, and underscores"),
+            );
+        }
```

Comment:
What would happen if this endpoint was submitted directly via the API? What makes file names outside of this pattern unsafe? What does safe mean in  this context?

```suggestion
        const ValidFileNamePattern = /^[a-zA-Z0-9._-]+$/;
        if (!ValidFileNamePattern.test(this.selectedFile.name)) {
            throw new Error(
                msg("Filename can only contain letters, numbers, dots, hyphens, and underscores"),
            );
        }
```